ci/base-images/cdxgen/alpine/Dockerfile.golang123

```diff
@@ -0,0 +1,50 @@
+FROM ghcr.io/cyclonedx/alpine-golang123:master
+
+LABEL maintainer="CycloneDX" \
+      org.opencontainers.image.authors="Team AppThreat <cloud@appthreat.com>" \
+      org.opencontainers.image.source="https://github.com/CycloneDX/cdxgen" \
+      org.opencontainers.image.url="https://github.com/CycloneDX/cdxgen" \
+      org.opencontainers.image.version="rolling" \
+      org.opencontainers.image.vendor="CycloneDX" \
+      org.opencontainers.image.licenses="Apache-2.0" \
+      org.opencontainers.image.title="cdxgen" \
+      org.opencontainers.image.description="Rolling image with cdxgen SBOM generator for dotnet 6 apps" \
+      org.opencontainers.docker.cmd="docker run --rm -v /tmp:/tmp -p 9090:9090 -v $(pwd):/app:rw -t ghcr.io/cyclonedx/cdxgen-debian-dotnet6:v11 -r /app --server"
+
+ENV CDXGEN_IN_CONTAINER=true \
+    NODE_COMPILE_CACHE="/opt/cdxgen-node-cache" \
+    PYTHONPATH=/opt/pypi:${PYTHONPATH} \
+    DOSAI_CMD=/usr/local/bin/dosai
+ENV PATH=${PATH}:/usr/local/bin:/opt/pypi/bin:/opt/cdxgen/node_modules/.bin:
+
+COPY . /opt/cdxgen
+
+RUN set -e; \
+    ARCH_NAME="$(dpkg --print-architecture)"; \
+    url=; \
+    case "${ARCH_NAME##*-}" in \
+        'amd64') \
+            DOSAI_ARCH_SUFFIX='-full'; \
+            ;; \
+        'x86_64') \
+            DOSAI_ARCH_SUFFIX='-full'; \
+            ;; \
+        'arm64') \
+            DOSAI_ARCH_SUFFIX='-linux-arm64-full'; \
+            ;; \
+        'aarch64') \
+            DOSAI_ARCH_SUFFIX='-linux-arm64-full'; \
+            ;; \
+        *) echo >&2 "error: unsupported architecture: '$ARCH_NAME'"; exit 1 ;; \
+    esac \
+    && cd /opt/cdxgen && corepack enable && corepack pnpm install --config.strict-dep-builds=true --prod --package-import-method copy --frozen-lockfile && corepack pnpm cache delete \
+    && curl -L https://github.com/owasp-dep-scan/dosai/releases/latest/download/Dosai${DOSAI_ARCH_SUFFIX} -o /usr/local/bin/dosai \
```

Comment:
There are two issues:
- dosai command is only required for dotnet images.
- Use the linux musl binaries for alpine. `Dosai-linux-musl-{amd64,arm64}` when the image already includes dotnet-sdk-9.0 or `Dosai-linux-musl-{amd64,arm64}-full` for use without the sdk. Insert `-musl` to the variable `DOSAI_ARCH_SUFFIX=`